psi4/src/psi4/lib3index/df_helper.cc

```diff
@@ -165,26 +176,54 @@ void DF_Helper::initialize() {
     prepare_sparsity();
     timer_off("DFH: sparsity prep");
 
-    // shut off AO core if necessary
-    if (memory_ * 0.8 < big_skips_[nao_]) {
-        AO_core_ = false;
-        outfile->Printf("\n    ==> DF_Helper memory announcement <==\n");
-        std::stringstream ann;
-        ann << "        Turning off in-core AOs.  DF_Helper needs at least (";
-        ann << big_skips_[nao_] / 0.9 * 8 / 1e9 << "GB), but it only got (" << memory_ * 8 / 1e9 << "GB)\n";
-        outfile->Printf(ann.str().c_str());
-    }
-    
+    // figure out AO_core
+    AO_core();   
+ 
     // prepare AOs for STORE method
-    if (AO_core_)
+    if (AO_core_) {
         prepare_AO_core();
-    else if (!direct_ && !direct_iaQ_)
+        if (do_wK_) { 
+            ; // TODO prepare_AO_wK_core();
+        }
+    } else if (!direct_ && !direct_iaQ_) {
         prepare_AO();
+        if (do_wK_) {
+            ; // TODO prepare_AO_wK();
+        }
+    }
 
     built_ = true;
     timer_off("DFH: initialize()");
+    
+    if(debug_) {
+        outfile->Printf("Exiting DF_Helper::initialize\n");
+    }
 }
+void DF_Helper::AO_core() {
 
+    // total size of sparse AOs
+    size_t required = ( do_wK_ ? 3 * big_skips_[nao_] : big_skips_[nao_]);
```

Comment:
On this note, do we raise anywhere if `do_wK_` is currently called?

Reply:
No, I set up the template for the wK code to be added, but there are currently no throws.  Shall I go ahead and do that? or are we planning to handle that logic elsewhere?

Reply:
This will be caught at other layers in typical workflows. But if its not implemented you should still raise an error as you are introducing an API that anyone can use.